cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/release.go

```diff
@@ -312,6 +315,51 @@ func (s *Server) helmReleaseFromUnstructured(ctx context.Context, name types.Nam
 	return release, nil
 }
 
+func (s *Server) newRelease(ctx context.Context, packageRef *corev1.AvailablePackageReference, targetName types.NamespacedName) (*corev1.InstalledPackageReference, error) {
+	// HACK: just for now assume HelmRelease CRD will live in the kubeapps namespace
+	kubeappsNamespace := os.Getenv("POD_NAMESPACE")
+	if kubeappsNamespace == "" {
+		// unit tests don't have this env var set
+		kubeappsNamespace = "kubeapps"
+	}
+	resourceIfc, err := s.getReleasesResourceInterface(ctx, kubeappsNamespace)
+	if err != nil {
+		return nil, err
+	}
+
+	availablePackageNamespace := packageRef.GetContext().GetNamespace()
+	if availablePackageNamespace == "" || packageRef.GetIdentifier() == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "required context or identifier not provided")
+	}
+
+	unescapedChartID, err := getUnescapedChartID(packageRef.Identifier)
+	if err != nil {
+		return nil, err
+	}
+
+	packageIdParts := strings.Split(unescapedChartID, "/")
+	repo := types.NamespacedName{Namespace: availablePackageNamespace, Name: packageIdParts[0]}
+	chart, err := s.fetchChartFromCache(repo, packageIdParts[1])
+	if err != nil {
+		return nil, err
+	}
+
+	fluxHelmRelease := newFluxHelmRelease(chart, kubeappsNamespace, targetName)
+	newRelease, err := resourceIfc.Create(ctx, fluxHelmRelease, metav1.CreateOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	name, err := namespacedName(newRelease.Object)
+	if err != nil {
+		return nil, err
+	}
+	return &corev1.InstalledPackageReference{
+		Context:    &corev1.Context{Namespace: name.Namespace},
+		Identifier: name.Name,
+	}, nil
+}
```

Comment:
Nice - that function is simpler than I expected it to be, due to the work you've done previously for `fetchChartFromCache` etc. :)

Reply:
thanks for the review.